packages/select/src/components/query-list/queryList.tsx

```diff
@@ -136,7 +136,7 @@ export class QueryList<T> extends React.Component<IQueryListProps<T>, IQueryList
             this.setState({ activeItem: nextProps.activeItem });
         }
         if (nextProps.query != null) {
-            this.setQuery(nextProps.query);
+            this.setQuery(nextProps.query, nextProps.resetOnQuery, nextProps);
```

Comment:
might this work better if it happened in `DidUpdate` instead? seems much better to invoke callbacks in a `Did` than in a `Will`.

Reply:
mm tho there is some state manipulation inside `setQuery` so it does make sense here.

Reply:
I'll leave it in componentWillRecieveProps for now as it's meant to be a simple bug fix. Moving things around in the lifecycle is probably a bigger job since things are getting deprecated for react V17